website/utils/url_utils.ts

```diff
@@ -139,6 +231,36 @@ export async function getArgsFromParams(urlSearchParams: URLSearchParams): Promi
     }
     args.viewerChannelSettings = initialChannelSettings;
   }
+  // Check for per-channel settings; this will override the old channel settings (`ch`)
+  // if present.
+  // Channels keys are formatted as `c0`, `c1`, etc., and the value is string containing
+  // a comma-separated list of key-value pairs.
+  const channelIndexToSettings: Map<number, ViewerChannelSetting> = new Map();
+  Object.keys(params).forEach((key) => {
+    if (CHANNEL_STATE_KEY_REGEX.test(key)) {
+      const channelIndex = parseInt(key.slice(1), 10);
+      try {
+        const channelData = parseKeyValueList(params[key]!);
+        const channelSetting = deserializeViewerChannelSetting(channelIndex, channelData as ViewerChannelSettingJson);
+        channelIndexToSettings.set(channelIndex, channelSetting);
+      } catch (e) {
+        console.warn(
```

Comment:
Not saying we should yet, but could we handle this as an error if we decided it was merited?

Reply:
Hmm, currently this lives outside of Cameron's error-handling code and alert banners, since this function is called by the wrapper around our viewer app. We'd have to pass in an error object as a prop or something to make it visible to a user, or copy the error component and have an instance of it live at a wrapper-level?